models/Schematic_FieldModel.php

```diff
@@ -0,0 +1,154 @@
+    {
+        return craft()->sections;
+    }
+
+    /**
+     * @return UserGroupsService
+     */
+    private function getUserGroupsService()
+    {
+        return craft()->userGroups;
+    }
+
+    /**
+     * @param FieldModel $field
+     * @param $includeContext
+     * @return array
+     */
+    public function getDefinition(FieldModel $field, $includeContext)
+    {
+        $definition = array(
+            'name' => $field->name,
+            'required' => $field->required,
+            'instructions' => $field->instructions,
+            'translatable' => $field->translatable,
+            'type' => $field->type,
+            'settings' => $field->settings,
+        );
+
+        if ($includeContext) {
+            $definition['context'] = $field->context;
+        }
+
+        if (isset($definition['settings']['sources'])) {
+            $definition['settings']['sources'] = $this->getMappedSources($definition['settings']['sources'], 'id', 'handle');
+        }
+
+        return $definition;
+    }
+
+    /**
+     * @param array $fieldDefinition
+     * @param FieldModel $field
+     * @param string $fieldHandle
+     * @param FieldGroupModel|null $group
+     */
+    public function populate(array $fieldDefinition, FieldModel $field, $fieldHandle, FieldGroupModel $group = null)
+    {
+        $field->name = $fieldDefinition['name'];
+        $field->handle = $fieldHandle;
+        $field->required = $fieldDefinition['required'];
+        $field->translatable = $fieldDefinition['translatable'];
+        $field->instructions = $fieldDefinition['instructions'];
+        $field->type = $fieldDefinition['type'];
+        $field->settings = $fieldDefinition['settings'];
+
+        if ($group) {
+            $field->groupId = $group->id;
+        }
+
+        if (isset($definition['settings']['sources'])) {
+            $settings = $fieldDefinition['settings'];
+            $settings['sources'] = $this->getMappedSources($settings['sources'], 'handle', 'id');
+            $field->settings = $settings;
+        }
+    }
+
+    /**
+     * Get sources based on the indexFrom attribute and return them with the indexTo attribute
+     *
+     * @param string|array $sources
+     *
+     * @param string $indexFrom
+     * @param string $indexTo
+     * @return array|string
+     */
+    private function getMappedSources($sources, $indexFrom, $indexTo)
+    {
+        if (!is_array($sources)) {
+            return $sources;
+        }
+
+        $mappedSources = array();
+        foreach ($sources as $source) {
+            $mappedSources[] = $this->getSource($source, $indexFrom, $indexTo);
+        }
+
+        return $mappedSources;
+    }
+
+    /**
+     * Gets a source by the attribute indexFrom, and returns it with attribute $indexTo
+     * @param string $source
+     * @param string $indexFrom
+     * @param string $indexTo
+     * @return string
+     */
+    private function getSource($source, $indexFrom, $indexTo)
```

Comment:
Even though this method is technically correct, you might want to split it up some more. It still seems to contain a bit too much logic for for my taste :-)

// Todo pherhaps?